docs/deploy/nginx-uwsgi.rst

```diff
@@ -0,0 +1,215 @@
+  wsgi-file = /home/myproject/src/wsgi.py
+  chdir = /home/myproject/src
+  uid = myproject-runner
+  gid = myproject-runner
+
+
+.. note:: Thread vs Processes vs gevent
+
+  There is a lot of questions to ask when deciding how to manage the processes
+  that actually run your Python code. Are you generally CPU bound or IO bound?
+  Is your application code thread safe? How many CPU's do you have? What system
+  are you on? Do you need an in-process cache?
+
+  You will have to do some research to understand this problem and no answers
+  can be given to your specific needs. In general uWSGI is so flexible you will
+  be able to find a solution that meets your applications needs. This
+  configuration enables both threads and processes. Reasearch, experiment, know
+  what you are doing, be smart.
+
+
+There are some important items in here, like ``uid`` and ``gid``. Note that
+those values are set to the application runner and not the owner of the
+application which should be the owner and group on the `/home/myproject/src` and
+`/home/myproject/venv` directory. You should have 755 permissions on the `src`
+and `venv` directories and 644 permission on the files within those directories.
+This should give you read-access for the `myproject-runner` user.
+
+You can now start uwsgi like this:
+
+.. code:: sh
+
+  $ /home/myproject/venv/bin/uwsgi -c uwsgi.ini
+
+If everything goes well you should see something like this:
+
+::
+
+    *** Operational MODE: preforking+threaded ***
+    WSGI app 0 (mountpoint='') ready in 1 seconds on interpreter 0x7fc5a282ba00 pid: 91828 (default app)
+    *** uWSGI is running in multiple interpreter mode ***
+    spawned uWSGI master process (pid: 91828)
+    spawned uWSGI worker 1 (pid: 91866, cores: 2)
+    spawned uWSGI worker 2 (pid: 91867, cores: 2)
+
+
+
+Making NGINX & uWSGI Talk
+'''''''''''''''''''''''''
+
+NGINX is used to proxy API requests to uWSGI which starts and manages your
+application. In NGINX parlance, we will create an "upstream" and direct that
+upstream (via local IP) to our now running uWSGI application.
+
+The configuration looks like this:
+
+.. code-block:: ini
+  :caption: /etc/nginx/sites-avaiable/myproject.conf
+
+  server {
+    listen 80;
+    server_name myproject.com;
+
+    access_log /var/log/nginx/myproject-access.log;
+    error_log  /var/log/nginx/myproject-error.log  warn;
+
+    location / {
+      uwsgi_pass 127.0.0.1:8080
+      include uwsgi_params;
+    }
+  }
+
+
+.. code-block:: sh
+
+  $ sudo service start nginx
+
+NGINX should start and you should now have a working application.
+
+
+Further Considerations
+''''''''''''''''''''''
+
+We did not explain how to setup TLS (HTTPS) for NGINX, that is an exercise for
+the reader. Consider using Let's Encrypt which offers free, short-term
+certificates which auto-renew the LE docs are the best place to learn how to
+integrate that within NGINX.
+
+Also, you might want to setup a database or any number of other services. That
```

Comment:
Change:

> Also, you might want to setup a database or any number of other services. That is not covered here since there are so many different services, possible configurations, network challenges, and security concerns.

To:

> In addition to setting up NGINX and uWSGI to run your application, you will of course need to deploy a database server or any other services required by your application. That being said, this guide does not cover the topic of ancillary services, since it involves so many different services, possible configurations, network challenges, and security concerns.